shared/api/image_transforms_qwen2_5.hpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cmath>
+#include <vector>
+#include "ext_status.h"
+#include "op_def_struct.h"
+
+namespace ort_extensions {
+
+class PatchImage {
+ public:
+  PatchImage() = default;
+
+  OrtxStatus Compute(const ortc::Tensor<float>& input, ortc::Tensor<float>& output) {
+    // Validate and read HWC
+    const auto& dims = input.Shape();
+    if (dims.size() != 3ULL) {
+      return {kOrtxErrorInvalidArgument, "[PatchImage]: input must be HWC float"};
+    }
+
+    int64_t H = dims[0];
+    int64_t W = dims[1];
+    int64_t C = dims[2];
+
+    const float* data = input.Data();
+    const int64_t total_elems = H * W * C;
+
+    // Convert RGB HWC -> CHW (channel 0 = R, 1 = G, 2 = B)
+    std::vector<float> chw;
+    chw.reserve(static_cast<size_t>(H) * static_cast<size_t>(W) * static_cast<size_t>(C));
+    for (int64_t ch = 0; ch < C; ++ch) {
+      for (int64_t h = 0; h < H; ++h) {
+        for (int64_t w = 0; w < W; ++w) {
+          size_t base = static_cast<size_t>((h * W + w) * C);
+          // Input is RGB so base+0=R, base+1=G, base+2=B
+          chw.push_back(data[base + ch]);
+        }
+      }
+    }
+
+    // Add batch dimension (frames = 1 initially) and prepare patches vector
+    std::vector<float> patches = chw;
```

Comment:
We shall avoid such vector copy. 

I think we can allocate with padded space. Fill them with padded value. Then apply HWC -> CHW (need to compute destination offset with padded height / width).